pkg/install/billing.go

```diff
@@ -21,10 +29,65 @@ func (i *Installer) createBillingRecord(ctx context.Context) error {
 			Location: i.doc.OpenShiftCluster.Location,
 		},
 	})
-	// If create return a conflict, this means row is already present in database
-	if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
-		return nil
+	if err != nil {
+		if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
+			i.log.Printf("billing record already present id DB")
+			return nil
+		}
+		return err
+	}
+	// Validate if E2E Feature is registred
+	resource, err := azure.ParseResourceID(billingDoc.Key)
+	if err != nil {
+		return err
 	}
 
-	return err
+	subDocument, err := i.sub.Get(ctx, resource.SubscriptionID)
+	if err != nil {
+		return err
+	}
+	if isSubscriptionRegisteredToE2E(*subDocument) {
+		// We are not failing the operation if we cannot write to e2e storage account, just warning
+		// The blob can only be created when the billing record is also written for the first time
+		errBlob := createE2Eblob(ctx, i.env, *billingDoc, resource)
+		if errBlob != nil {
+			i.log.Warnf("createE2Eblob failed : %s", errBlob.Error())
+		}
+	}
+	return nil
+}
+
+func createE2Eblob(ctx context.Context, env env.Interface, doc api.BillingDocument, resource azure.Resource) error {
+	client, err := azstorage.NewBasicClient(env.E2EStorageAccountName(), env.E2EStorageAccountKey())
+	if err != nil {
+		return err
+	}
+	bs := client.GetBlobService()
+	// Create container if it does not exist: [bill]-[region]-[resourceGroupName]-[resourceName]
+	containerName := strings.ToLower("bill-" + doc.Billing.Location + "-" + resource.ResourceGroup + "-" + resource.ResourceName)
+	if len(containerName) > 63 {
+		containerName = containerName[:62] // maximum characters allowed is 63
+	}
+	containerRef := bs.GetContainerReference(containerName)
+	_, err = containerRef.CreateIfNotExists(nil)
+	if err != nil {
+		return fmt.Errorf("Error creating container : %s", err.Error())
+	}
+	blobRef := containerRef.GetBlobReference("billingentity")
+	b, err := json.Marshal(doc)
+	if err != nil {
+		return err
+	}
+	return blobRef.CreateBlockBlobFromReader(bytes.NewReader(b), nil)
+}
+
+// isSubscriptionRegisteredToE2E returns true if the subscription is having the
+// "Microsoft.ContainerService/SaveAROTestConfig" feature registered
+func isSubscriptionRegisteredToE2E(doc api.SubscriptionDocument) bool {
+	for _, feature := range doc.Subscription.Properties.RegisteredFeatures {
+		if feature.Name == api.SubscriptionFeatureE2E && feature.State == "Registered" {
+			return true
+		}
```

Comment:
To be safe should it also validate if the sub is in AME or MSFT tenant? @shrutir25 thoughts?

Reply:
@amanohar we typically use a MSFT sub in INT and an AME sub in Prod, so we can add validation for the environment if needed.

Reply:
added